spec/requests/listings_spec.rb

```diff
@@ -83,19 +86,20 @@
   end
 
   describe "PATCH /update" do
-    let(:listing) { create(:listing) }
+    let(:listing) { create(:listing, :with_contact_info) }
 
     context "with valid parameters" do
+      let(:new_street_address) { Faker::Address.street_address }
       let(:new_attributes) {{
-        location_attributes: {street_address: 'changed'},
+        location_attributes: {street_address: new_street_address, zip: Faker::Address.zip(state_abbreviation: 'MI')},
```

Comment:
Passing `state_abbreviation` here is a kludge to coerce Faker into not trying zip+4 values

I was going to make the form allow zip+4, using the regex from https://github.com/dgilperez/validates_zipcode/blob/6345983cb72dd3a2776ca06a42039be7df75d307/lib/validates_zipcode/cldr_regex_collection.rb#L15. The test still failed due to DB limitations, so this is something we can revisit later